common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -181,6 +182,7 @@ def publish_event(event_name, **kwargs):
         if overlimit_block_keys:
             # Publish an event for analytics purposes:
             publish_event("removed", removed=format_block_keys(overlimit_block_keys), reason="overlimit")
+            previous_count = len(selected)
```

Comment:
Also, is it the case that `previous_count` should always be the length of `select` from the last time that the `publish_event` method was called? If so, I'd advocate moving `publish_event` to be a method on the class, and storing the length of `selected` passed into it, and using that stored value the next time an event is published. That way, you don't have to make sure that you're always updating the variables in tandem.